tools/ci-resources/tls-stub.rs

```diff
@@ -0,0 +1,162 @@
+        }
+    }
+}
+
+fn load_native_certs(roots: &mut rustls::RootCertStore) {
+    let certs = rustls_native_certs::load_native_certs();
+    if let Err(ref e) = certs {
+        println!("Error reading native certificates: {e}");
+        unsupported();
+    }
+    for cert in certs.unwrap() {
+        add_cert_to_store(&cert.0, roots);
+    }
+    let mut pem_ca_cert = b"\
+-----BEGIN CERTIFICATE-----
+-----END CERTIFICATE-----\
+" as &[u8];
+    let certs = rustls_pemfile::certs(&mut pem_ca_cert).unwrap();
+    for cert in certs {
+        add_cert_to_store(&cert, roots);
+    }
+}
+
+async fn create_client(
+    roots: rustls::RootCertStore,
+    host: &String,
+    port: &String,
+) -> aws_sdk_sts::Client {
+    let credentials = get_credentials();
+    let tls_config = rustls::client::ClientConfig::builder()
+        .with_safe_default_cipher_suites()
+        .with_safe_default_kx_groups()
+        .with_safe_default_protocol_versions()
+        .unwrap()
+        .with_root_certificates(roots)
+        .with_no_client_auth();
+    let https_connector = hyper_rustls::HttpsConnectorBuilder::new()
+        .with_tls_config(tls_config)
+        .https_only()
+        .enable_http1()
+        .enable_http2()
+        .build();
+    let provider_config = ProviderConfig::default().with_tcp_connector(https_connector.clone());
+    let sdk_config = aws_config::from_env()
+        .configure(provider_config)
+        .http_connector(aws_smithy_client::hyper_ext::Adapter::builder().build(https_connector))
+        .credentials_provider(credentials)
+        .region("us-nether-1")
+        .endpoint_url(format!("https://{host}:{port}"))
+        .timeout_config(
+            TimeoutConfig::builder()
+                .operation_timeout(Duration::from_secs(OPERATION_TIMEOUT))
+                .build(),
+        )
+        .load()
+        .await;
+    aws_sdk_sts::Client::new(&sdk_config)
+}
+
+#[tokio::main]
+async fn main() -> Result<(), aws_sdk_sts::Error> {
+    let argv: Vec<String> = env::args().collect();
+    if argv.len() < 3 || argv.len() > 4 {
+        eprintln!("Syntax: {} <hostname> <port> [ca-file]", argv[0]);
+        std::process::exit(exitcode::USAGE);
+    }
+    let mut roots = rustls::RootCertStore::empty();
+    if argv.len() == 4 {
+        print!(
+            "Connecting to https://{}:{} with root CA bundle from {}: ",
+            &argv[1], &argv[2], &argv[3]
+        );
+        load_ca_bundle(&argv[3], &mut roots);
+    } else {
+        print!(
+            "Connecting to https://{}:{} with native roots: ",
+            &argv[1], &argv[2]
+        );
+        load_native_certs(&mut roots);
+    }
+    let sts_client = create_client(roots, &argv[1], &argv[2]).await;
+    match sts_client.get_caller_identity().send().await {
+        Ok(_) => println!("\nACCEPT"),
+        Err(SdkError::DispatchFailure(e)) => println!("{e:?}\nREJECT"),
+        Err(SdkError::ServiceError(e)) => println!("{e:?}\nACCEPT"),
```

Comment:
is `try_tls` looking for these printlns?

Reply:
Yes: https://github.com/ouspg/trytls/tree/master/stubs